### PR TITLE
Added building SparseSimulator without Cake

### PR DESCRIPTION
**Pull request by kuzminrobin**
Also replaced `cl.exe` compiler on Win with Clang.
It is not to `main` it is t o a feature branch.

### CODE REVIEW

File: src/Simulation/Simulators/SparseSimulator/build.ps1 [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        Write-Host "On MacOS build using the default C/C++ compiler (should be AppleClang)"
+    }
+    else {
+        if (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
+        {
+            Write-Host "On Linux build using Clang"
+            $CC = "clang-11"
+            $CXX = "clang++-11"
+            #$clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy-11"
```
Comment (kuzminrobin) [inline src/Simulation/Simulators/SparseSimulator/build.ps1:36]: I would like to preserve the `clangTidy` fragments for now.

File: src/Simulation/Simulators/SparseSimulator/build.ps1 [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    cmake --build .
+
+# popd
+Pop-Location
+
+
+# BUILD C# PART AND TESTS
+
+dotnet build . --configuration $Env:BUILD_CONFIGURATION
+# dotnet test . --configuration $Env:BUILD_CONFIGURATION
```
Comment (kuzminrobin) [inline src/Simulation/Simulators/SparseSimulator/build.ps1:73]: I would like to preserve this commented-out command for now.

File: src/Simulation/Simulators/SparseSimulator/build.ps1 [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                # LLVM was installed by Chocolatey, so add the install location to the path.
+                $env:PATH += ";$($env:SystemDrive)\Program Files\LLVM\bin"
+            }
+
+            #if (Get-Command clang-tidy -ErrorAction SilentlyContinue) {
+            #    # Only run clang-tidy if it's installed. This is because the package used by chocolatey on
+            #    # the build pipeline doesn't include clang-tidy, so we allow skipping that there and let
+            #    # the Linux build catch tidy issues.
+            #    $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy"
+            #}
```
Comment (kuzminrobin) [inline src/Simulation/Simulators/SparseSimulator/build.ps1:54]: I would like to preserve the `clangTidy` fragments for now.

**Comment by kuzminrobin** [comment]
Please review.

File: bootstrap.ps1 [hunk 4]
```diff
@@ -21,8 +21,13 @@ Pop-Location
...
 if (-not (Test-Path Env:/AGENT_OS)) {                                    # If not CI build, i.e. local build (if AGENT_OS envvar is not defined)
     if ($Env:ENABLE_NATIVE -ne "false") {
-        Write-Host "Build release flavor of the native simulator"
         $Env:BUILD_CONFIGURATION = "Release"
+        Write-Host "Build release flavor of the Sparse Simulator"
+        Push-Location (Join-Path $PSScriptRoot "src/Simulation/Simulators/SparseSimulator")
+            .\build.ps1
+        Pop-Location
+
+        Write-Host "Build release flavor of the native simulator"
```
Comment (DmitryVasilevsky) [inline bootstrap.ps1:30]: I suggest you change it to "full state".

File: src/Simulation/Simulators/SparseSimulator/build.ps1 [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+# mkdir Native\build\(Debug|Release)
+$BuildDir = (Join-Path $PSScriptRoot "Native" "build" $Env:BUILD_CONFIGURATION)
+if (-not (Test-Path $BuildDir)) {
+    New-Item -Path $BuildDir -ItemType "directory" | Out-Null
+}
+
+# pushd Native\build\(Debug|Release)
+Push-Location $BuildDir
+
+    $CmakeConfigCommand = "& cmake -G Ninja -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON -D CMAKE_BUILD_TYPE=$Env:BUILD_CONFIGURATION -S ..\.. "  # Without `-G Ninja` the compiler chosen is always `cl.exe`.
```
Comment (DmitryVasilevsky) [inline src/Simulation/Simulators/SparseSimulator/build.ps1:20]: Why does it start with '&'?

**Review by DmitryVasilevsky — APPROVED** [review]

File: src/Simulation/Simulators/SparseSimulator/build.ps1 (hunk 5 above)
Comment (kuzminrobin) [reply src/Simulation/Simulators/SparseSimulator/build.ps1]: Removed.

File: bootstrap.ps1 (hunk 4 above)
Comment (kuzminrobin) [reply bootstrap.ps1]: Changed.

**Pull request merged**